sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -700,14 +723,38 @@ case class HashShuffleSpec(
     }
   }
 
-  override def createPartitioning(clustering: Seq[Expression]): Partitioning = {
+  override def createPartitioning(clustering: Seq[Expression]): HashPartitioning = {
```

Comment:
Why do we need this change ?

Reply:
Not needed, just me being sloppy when trying out some other approach. reverted.